### PR TITLE
anndata: Add tools to manage AnnData datasets

### PR DESCRIPTION
**Pull request by bebatut**
FOR CONTRIBUTOR:
* [X] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [X] - License permits unrestricted use (educational + commercial)
* [X] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

It used the h5ad format which has been added in https://github.com/galaxyproject/galaxy/pull/7268 (so it will fail with the current used version of Galaxy).

### CODE REVIEW

File: tools/anndata/import.xml [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+&&
+cp '$in.matrix' mtx/matrix.mtx.gz
+&&
+cp '$in.tenx.features' mtx/features.tsv.gz
+&&
+cp '$in.tenx.barcodes' mtx/barcodes.tsv.gz
+    #end if
+#else if $in.adata_format == 'umi_tools'
+&&
+gunzip '$in.input'
```
Comment (bgruening) [inline tools/anndata/import.xml]: in.input is just a tabular isn't it? Do you want to zip it instead?

File: tools/anndata/import.xml [hunk 2]
```diff
@@ -0,0 +1,234 @@
+<tool id="anndata_import" name="Import AnnData" version="@VERSION@+@GALAXY_VERSION@">
+    <description>from different format</description>
+    <macros>
+        <import>macros.xml</import>
+        <xml name="params_10x">
+            <param name="barcodes" type="data" format="tabular" label="Barcodes"/>
```
Comment (bgruening) [inline tools/anndata/import.xml]: How does this file looks like? Two columns, or one? If one maybe tabular and txt can work both?

File: tools/anndata/import.xml [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+                <param name="first_column_names" value="true"/>
+            </conditional>
+            <assert_stdout>
+                <has_text_matching expression="ad.read_csv"/>
+                <has_text_matching expression="delimiter='\\t'"/>
+                <has_text_matching expression="first_column_names=True"/>
+            </assert_stdout>
+            <output name="anndata" value="import.tsv.h5ad" ftype="h5ad" compare="sim_size"/>
+        </test>
+        <!--<test>
```
Comment (bgruening) [inline tools/anndata/import.xml]: Should we include these tests before merging?

File: tools/anndata/manipulate.xml [hunk 4]
```diff
@@ -0,0 +1,175 @@
+<tool id="anndata_manipulate" name="Manipulate AnnData" version="@VERSION@+@GALAXY_VERSION@">
+    <description>object</description>
+    <macros>
+        <import>macros.xml</import>
+        <xml name="param_join">
+            <param name="join" type="text" value="-" label="The connecting string between name and integer"/>
```
[the remark this answers is not in the document]
Comment (bebatut) [reply tools/anndata/manipulate.xml]: I replaced it by a select with some possible characters

File: tools/anndata/import.xml (hunk 2 above)
Comment (bebatut) [reply tools/anndata/import.xml]: I added txt also here

File: tools/anndata/import.xml (hunk 3 above)
Comment (bebatut) [reply tools/anndata/import.xml]: It would be nice. But we need https://github.com/galaxyproject/galaxy/pull/7569 then

**Comment by bebatut** [comment]
I expanded the help section also
Thanks @bgruening for the comments

**Comment by bgruening** [comment]
Awesome!

**Pull request merged**